wrapper/src/main/java/software/amazon/jdbc/PropertyDefinition.java

```diff
@@ -95,7 +95,20 @@ public static Collection<AwsWrapperProperty> allProperties() {
     return PROPS_BY_NAME.values();
   }
 
-  public static void removeAll(Properties props) {
-    PROPS_BY_NAME.keySet().forEach(props::remove);
+  public static void removeAllExceptCredentials(Properties props) {
+    String user = props.containsKey(PropertyDefinition.USER.name)
+        ? props.getProperty(PropertyDefinition.USER.name) : null;
+    String password = props.containsKey(PropertyDefinition.PASSWORD.name)
+        ? props.getProperty(PropertyDefinition.PASSWORD.name) : null;
+
+    props.clear();
```

Comment:
It should be 
PROPS_BY_NAME.keySet().forEach(props::remove);

Reply:
Ah good catch! Fixed!